src/chat.rs

```diff
@@ -3963,6 +4022,41 @@ pub(crate) async fn update_msg_text_and_timestamp(
     Ok(())
 }
 
+impl Context {
+    /// Executes [`sync::AlterChat`] item sent by other device.
+    pub(crate) async fn sync_alter_chat(&self, data: &sync::AlterChatData) -> Result<()> {
+        let chat_id = match &data.id {
+            sync::ChatId::ContactAddr(addr) => {
+                let Some(contact_id) =
+                    Contact::lookup_id_by_addr_ex(self, addr, Origin::Unknown, None).await?
+                else {
+                    warn!(self, "sync_alter_chat: No contact for addr '{addr}'.");
+                    return Ok(());
+                };
+                let Some(chat_id) = ChatId::lookup_by_contact(self, contact_id).await? else {
+                    warn!(self, "sync_alter_chat: No chat for addr '{addr}'.");
+                    return Ok(());
+                };
+                chat_id
+            }
+            sync::ChatId::Grpid(grpid) => {
+                let Some((chat_id, ..)) = get_chat_id_by_grpid(self, grpid).await? else {
+                    warn!(self, "sync_alter_chat: No chat for grpid '{grpid}'.");
+                    return Ok(());
+                };
+                chat_id
+            }
+        };
+        match data.action {
+            sync::ChatAction::Block => chat_id.block(self).await,
```

Comment:
I'm unsure it will be actually simpler and more reliable as it would create another one way to modify the db. In the current approach it's the only check of `is_sync` per an API function. Even if some other API functions are called internally (which is unlikely), it isn't a recursion, but some "back sync". If we sync the db state instead, then it would be a separate code for each entity to sync and even more code as a result. And also we should care about db consistency, maybe it's easy for chats' `Blocked` state, but i'm unsure about all things we want to sync. Sanity and consistency checks would be bypassed on "slave" devices and for me it looks dangerous. There can be different versions of DC on devices and differences (and also bugs) in the sanity checks (or configs) on one device could lead to inconsistent values on other ones. Better if a user action isn't executed on a "slave" at all if the logic differs from the one on the "master" for any reason. Further, proper events should be generated in case of "state sync", and maybe other side effects should happen.

Reply:
So, all the "infrastructure" code is already written and i don't expect more complexity here when implementing synchronisation of other entities. Problems may happen if we have top-level API somewhere which looks complicated like in the case of unblocking a chat that should be done by unblocking a contact instead. But maybe it's a reason to think of our API once again.

Reply:
avoiding a new global state that potetially needs to be in mind everywhere sounds better, indeed.  even if some code to handle global-is-sync was already written - we are usually never afraid of throwing away code :)

if we really find a flag useful for some sync items in the future (so using the same function for update+send-sync and update-only, we can always pass a flag as a paramete - or split funktions accordingly (as for blocking already now - we could just call set_blocked())

Reply:
`is_sync` -- renamed to `nosync` now -- exists in the code not only to avoid a recursive sync, but also an unintended one like in `ChatId::create_for_contact_with_blocked()` which calls `ChatId::unblock()`, otherwise it will try to unblock on other device a chat that doesn't yet exist. So, even if we follow @link2xt's suggestion to sync a state instead of user actions, this `nosync` flag would remain. But this flag of course can be replaced with additional function parameters -- we can add a layer behind of our public API and add any parameters we need. But i decided to add it to `Context` instead because the device sync looks like a feature permeating a significant part of DC functionality and having a separate flag in lots of functions instead and forwarding this flag between them could grow the code significantly. If i'm wrong and there are not lots of places in the code where we should track this flag, we can always refactor this.

> i can already see copy+pasted code that maintains is_sync branches that are never needed, but there "to be sure" :)

Now this `nosync` flag is checked only in the places calling `add_sync_item()` and i don't foresee other reasons to check it. So, even if the code is copy+pasted, but it doesn't perform the device sync, i.e. `add_sync_item()` calls are removed then, the checks of `nosync` are removed too.

I'd suggest to postpone refactoring, i.e. replacing `Context::nosync` with a set of functions taking an additional parameter, until it's clear how many things we want to sync potentially. If these are only states of chats and contacts, we can do it easily as there's not much code to refactor.